storage-proofs/src/piece_inclusion_proof.rs

```diff
@@ -90,10 +133,44 @@ fn generate_piece_commitment<H: Hasher>(data: &[u8]) -> Result<H::Domain> {
 
 #[allow(dead_code)]
 /// Generate `comm_p` from bytes and return it as bytes.
-fn generate_piece_commitment_bytes<H: Hasher>(data: &[u8]) -> Result<Fr32Vec> {
+fn generate_piece_commitment_bytes<H: Hasher>(data: &[u8]) -> Result<Fr32Ary> {
     let comm_p = generate_piece_commitment::<H>(data)?;
+    let mut comm_p_bytes: Fr32Ary = [0; 32];
+
+    comm_p.write_bytes(&mut comm_p_bytes)?;
```

Comment:
BLOCKING: What is the purpose of this `comm_p.write_bytes(&mut comm_p_bytes)?;` call?